ios-target-app/Extensions/ButtonExtension.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  ButtonExtension.swift
+//  ios-target-app
+//
+//  Created by Raul Piñeres Carrera on 9/09/22.
+//
+
+import Foundation
+import UIKit
+
+extension UIButton {
+    
+    convenience init(
+        style: ButtonStyle,
+        actionHandler: (target: Any, action: Selector)? = nil
+    ) {
+        self.init()
+        
+        translatesAutoresizingMaskIntoConstraints = false
+        
+        if let action = actionHandler?.action {
+            addTarget(actionHandler?.target, action: action, for: .touchUpInside)
+        }
+        
+        switch style {
+        case .primary(let color, let title, let titleColor, let cornerRadius, let height):
+            setTitle(title, for: .normal)
+            setTitleColor(titleColor, for: .normal)
+            backgroundColor = color
+            setRoundBorders(cornerRadius)
+            heightAnchor.constraint(equalToConstant: height).isActive = true
+        case .secondary(let color, let title, let titleColor, let cornerRadius, let height):
+            setTitle(title, for: .normal)
+            setTitleColor(titleColor, for: .normal)
+            backgroundColor = color
+            setRoundBorders(cornerRadius)
+            heightAnchor.constraint(equalToConstant: height).isActive = true
+        }
```

Comment:
Instead of having each ButtonStyle case to have attributes, the enum can have computed properties that are determined based on the case. You could also have a `.custom(color:,...)` case
This way you avoid repeating this blocks for each case:

```suggestion
      case .custom(let color, let title, let titleColor, let cornerRadius, let height):
            setTitle(title, for: .normal)
            setTitleColor(titleColor, for: .normal)
            backgroundColor = color
            setRoundBorders(cornerRadius)
            heightAnchor.constraint(equalToConstant: height).isActive = true
       default:
            setTitle(style.title, for: .normal)
            setTitleColor(style.titleColor, for: .normal)
            backgroundColor = style.color
            setRoundBorders(style.cornerRadius)
            heightAnchor.constraint(equalToConstant: style.height).isActive = true
      }
```

Reply:
i'm agree with @glm4. Also other alternative is to extract that repeated code in a single func, then call this func with the custom parameters needed. I'd keep the .primary/.secondary but in a single line implementing the mentioned before if this styling is used in many places. Also i'd keep the .custom recommended below 💪